src/components/pill-selector-dropdown/Pill.js

```diff
@@ -16,12 +17,14 @@ const Pill = ({ isDisabled = false, isSelected = false, isValid = true, onRemove
         'is-invalid': !isValid,
         'is-disabled': isDisabled,
     });
+    const onClick = isDisabled ? noop : onRemove;
+
     return (
         <span className={styles}>
             <span className="pill-text">{text}</span>
-            <button aria-hidden="true" className="close-btn" disabled={isDisabled} onClick={onRemove} type="button">
+            <span aria-hidden="true" className="close-btn" onClick={onClick}>
```

Comment:
This element should use something other than a `span`, as they're not intended to be interactive and carry little semantic meaning.

Reply:
Agreed but I think the best candidate is `button` which is the element that introduced the bug. There's a deeper issue that's causing this element to get clicked when the parent is clicked, but I haven't been able to figure it out yet. This update is just meant to be a temporary workaround for now

Reply:
Fair enough. Let's come back to it in the future.